exporter/opentelemetry-exporter-zipkin/src/opentelemetry/exporter/zipkin/encoder/__init__.py

```diff
@@ -0,0 +1,267 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Zipkin Exporter Transport Encoder
+
+Base module and abstract class for concrete transport encoders to extend.
+"""
+
+import abc
+import json
+import logging
+from enum import Enum
+from typing import Any, Dict, List, Optional, Sequence, TypeVar
+
+from opentelemetry.exporter.zipkin.node_endpoint import NodeEndpoint
+from opentelemetry.sdk.trace import Event
+from opentelemetry.trace import Span, SpanContext
+from opentelemetry.trace.status import StatusCode
+
+EncodedLocalEndpointT = TypeVar("EncodedLocalEndpointT")
+
+DEFAULT_MAX_TAG_VALUE_LENGTH = 128
+NAME_KEY = "otel.library.name"
+VERSION_KEY = "otel.library.version"
+
+logger = logging.getLogger(__name__)
+
+
+class Encoding(Enum):
+    """Enum of supported encoding formats.
+
+    Values are human-readable strings so that they can be easily used by the
+    OS environ var OTEL_EXPORTER_ZIPKIN_ENCODING.
```

Comment:
Good catch. To fully align with spec I:
- updated env var name and noted that its reserved for future usage
- renamed all `encoding` classes / vars to be `protocol`
- made `protocol` a mandatory arg in exporter init since there is no defined default